bindings/sundials4py/examples/cvodes/cvs_brusselator.py

```diff
@@ -0,0 +1,200 @@
+# Newton iteration with the SUNDIALS dense linear solver, and a
+# user-supplied Jacobian routine.
+#
+# 100 outputs are printed at equal intervals, and run statistics
+# are printed at the end.
+# -----------------------------------------------------------------
+
+import sys
+import numpy as np
+from sundials4py.core import *
+from sundials4py.cvodes import *
+
+
+class BrusselatorODE:
+    def __init__(self, u0, v0, w0, a, b, ep):
+        self.u0 = u0
+        self.v0 = v0
+        self.w0 = w0
+        self.a = a
+        self.b = b
+        self.ep = ep
+
+    def set_init_cond(self, yvec):
+        y = N_VGetArrayPointer(yvec)
+        y[0] = self.u0
+        y[1] = self.v0
+        y[2] = self.w0
+        return 0
+
+    def f(self, t, yvec, ydotvec):
+        y = N_VGetArrayPointer(yvec)
+        ydot = N_VGetArrayPointer(ydotvec)
+        a, b, ep = self.a, self.b, self.ep
+        u, v, w = y[0], y[1], y[2]
+        ydot[0] = a - (w + 1.0) * u + v * u * u
+        ydot[1] = w * u - v * u * u
+        ydot[2] = (b - w) / ep - w * u
+        return 0
+
+    def jac(self, t, yvec, fyvec, J, tmp1, tmp2, tmp3):
+        y = N_VGetArrayPointer(yvec)
+        a, b, ep = self.a, self.b, self.ep
+        u, v, w = y[0], y[1], y[2]
+        Jdata = SUNDenseMatrix_Data(J)
+        Jdata[0, 0] = -(w + 1.0) + 2.0 * u * v
+        Jdata[0, 1] = u * u
+        Jdata[0, 2] = -u
+        Jdata[1, 0] = w - 2.0 * u * v
+        Jdata[1, 1] = -u * u
+        Jdata[1, 2] = u
+        Jdata[2, 0] = -w
+        Jdata[2, 1] = 0.0
+        Jdata[2, 2] = -1.0 / ep - u
+        return 0
+
+
+def main():
+    # Problem parameters for Test 2
+    u0 = 1.2
+    v0 = 3.1
+    w0 = 3.0
+    a = 1.0
+    b = 3.5
+    ep = 5.0e-6
+    T0 = 0.0
+    Tf = 10.0
+    dTout = 1.0
+    NEQ = 3
+    Nt = int(np.ceil(Tf / dTout))
+    reltol = 1.0e-6
+    abstol = 1.0e-10
+
+    status, sunctx = SUNContext_Create(SUN_COMM_NULL)
+    y = N_VNew_Serial(NEQ, sunctx)
+
+    ode_problem = BrusselatorODE(u0, v0, w0, a, b, ep)
+    ode_problem.set_init_cond(y)
+
+    cvode = CVodeCreate(CV_BDF, sunctx)
+    status = CVodeInit(cvode.get(), lambda t, y, ydot, _: ode_problem.f(t, y, ydot), T0, y)
+    assert status == CV_SUCCESS
+    status = CVodeSStolerances(cvode.get(), reltol, abstol)
+    assert status == CV_SUCCESS
+    status = CVodeSetMaxNumSteps(cvode.get(), 10000)
+    assert status == CV_SUCCESS
+
+    # Dense matrix and linear solver
+    A = SUNDenseMatrix(NEQ, NEQ, sunctx)
+    LS = SUNLinSol_Dense(y, A, sunctx)
+    status = CVodeSetLinearSolver(cvode.get(), LS, A)
+    assert status == CV_SUCCESS
+    status = CVodeSetJacFn(
+        cvode.get(),
+        lambda t, yvec, fyvec, J, tmp1, tmp2, tmp3, _: ode_problem.jac(
+            t, yvec, fyvec, J, tmp1, tmp2, tmp3
+        ),
+    )
+    assert status == CV_SUCCESS
+
+    # Parse any command line arguments
+    status = CVodeSetOptions(cvode.get(), "", "", len(sys.argv), sys.argv)
```

Comment:
Have you tested out whether command-line control works so easily in Python?  If so, this is outstanding!

Reply:
Here is the code that makes it work: https://github.com/LLNL/sundials/blob/feature/python-nanobind/bindings/sundials4py/cvodes/cvodes.cpp#L101.